sdks/python/apache_beam/yaml/yaml_join.py

```diff
@@ -62,9 +62,12 @@ def _validate_equalities(equalities, pcolls):
   error_prefix = f'Invalid value "{equalities}" for "equalities".'
 
   valid_cols = {
-      name: set(dict(pcoll.element_type._fields).keys())
+      name: set(
+          dict(fields).keys() if fields and all(
+              isinstance(field, tuple) for field in fields) else fields)
       for name,
       pcoll in pcolls.items()
```

Comment:
Until yapf fixes its bug, put ()'s around (name, pcoll) to get better wrapping.

Reply:
done